mlflow/gateway/providers/cohere.py

```diff
@@ -45,6 +46,63 @@ def model_to_completions(cls, resp, config):
             ),
         )
 
+    @classmethod
+    def model_to_completions_streaming(cls, resp, config):
+        # Response example (https://docs.cohere.com/reference/generate)
+        #
+        # Streaming chunks:
+        # ```
+        # {"index":0,"text":" Hi","is_finished":false,"event_type":"text-generation"}
+        # ```
+        # ```
+        # {"index":1,"text":" Hi","is_finished":false,"event_type":"text-generation"}
+        # ```
+        # notes: "index" is only present if "num_generations" > 1
+        #
+        # Final chunk:
+        # ```
+        # {"is_finished":true,"event_type":"stream-end","finish_reason":"COMPLETE",
+        #   "response":{"id":"b32a70c5-8c91-4f96-958f-d942801ed22f",
+        #       "generations":[
+        #           {
+        #               "id":"5d5d0851-35ac-4c25-a9a9-2fbb391bd415",
+        #               "index":0,
+        #               "text":" Hi there! How can I assist you today? ",
+        #               "finish_reason":"COMPLETE"
+        #           },
+        #           {
+        #               "id":"0a24787f-504e-470e-a088-0bf801a2c72d",
+        #               "index":1,
+        #               "text":" Hi there, how can I assist you today? ",
+        #               "finish_reason":"COMPLETE"
+        #           }
+        #       ],
+        #       "prompt":"Hello"
+        #   }}
+        # ```
+        response = resp.get("response")
+        return completions.StreamResponsePayload(
+            id=response["id"] if response else None,
```

Comment:
Is `id` only present in the final chunk?

Reply:
yes